Draft: Add PostgresDB backend support for Lighthouse Beacon Node

owanikin opened the pull request:
## Issue #7638 

## Proposed Changes

- Add initial scaffolding for postgres_impl.rs based on leveldb_impl.rs
- Setup feature flag in Cargo.toml for Postgres backend
- Add placeholder KeyValueStore trait impl using sqlx
- Integration points for Beacon Node backend to support Postgres

## Additional Info
- This is an early draft PR meant to gather feedback on approach, structure, and trait compatibility.

michaelsproul committed:
Change default state cache size back to 128 (#7364)

Closes:

- https://github.com/sigp/lighthouse/issues/7363


  - Change default state cache size back to 128.
- Make state pruning properly LRU rather than MSU after skipping the cull-exempt states.

macladson committed:
Release v7.0.1 (#7374)

Bump all required version numbers to `v7.0.1` to prepare for the next release.

jimmygchen committed:
Add additional mergify rules to automate triaging (#7451)

* Add additional mergify rules to automate triaging.

* Update mergify config.

jimmygchen committed:
Fix mergify infinite loop. (#7463)

* Fix mergify infinite loop.

* Update rule for `ready-for-review` label.

* More fix to prevent infinite loop

jimmygchen committed:
Prevent mergify from updating labels while CI is still running. (#7470)

jimmygchen committed:
One more attempt to fix mergify condition. (#7472)

jimmygchen committed:
Update mergify rule to not evaluate PRs that are not ready for review - to reduce noise and avoid updating stale PRs. (#7494)

jimmygchen committed:
Update mergify rules so that I can add `waiting-on-author` on a PR that's passing CI. Remove noisy comments.

jimmygchen committed:
Fix condition

jimmygchen committed:
Remove redundant `and`

eserilev commented:
I know its just an early draft PR but I think you'll need tor rebase your branch to unstable

owanikin commented:
> I know its just an early draft PR but I think you'll need tor rebase your branch to unstable

it says `
Current branch add-postgres-db is up to date.` after I `git rebase upstream/unstable` from my end, so i think its already rebased, or am I missing something?

eserilev committed:
Remove instrumenting log level  (#7636)

https://github.com/sigp/lighthouse/issues/7155


  Theres some additional places we set instrumenting log levels that wasn't covered in #7620

eserilev commented:
I think your branch has some changes from `stable`

owanikin committed:
feat(store): Add PostgresDB backend with basic get/put and tests

owanikin commented:
> I think your branch has some changes from `stable`

I’ve pushed a [WIP](https://github.com/sigp/lighthouse/pull/7639/commits/00856366bbf5ec551ac7a90f6e94cc3a892f82ac) PostgresDB implementation (get_bytes, put_bytes) with a test. Looking for feedback before wiring it in.

eserilev commented:
The PR diff is a bit messy You're reverting a bunch of changes from unstable so its hard to review this

owanikin commented:
> The PR diff is a bit messy You're reverting a bunch of changes from unstable so its hard to review this

@eserilev  I’ve pushed a clean version of the PostgresDB backend implementation (with get_bytes, put_bytes, and basic tests) in the [new PR](https://github.com/sigp/lighthouse/pull/7685). Not sure if you've had a chance to see it yet, but I'd really appreciate any early feedback on the direction before wiring it into the broader store interface. Thanks!

eserilev commented:
closing in favor of https://github.com/sigp/lighthouse/pull/7685
